### PR TITLE
Tests for evennia/utils/validatorfuncs.

### PR DESCRIPTION
**Pull request by Henddher**
Tests for evennia/utils/validatorfuncs.

- text()
- color()
- datetime()
- duration()
- future()
- signed_integer()
- positive_integer()
- unsigned_integer()
- boolean()
- timezone()
- email()
- lock()

Coverage on this module increased from 15.87% to 100%.

#1458 Hacktoberfest

### CODE REVIEW

File: evennia/utils/validatorfuncs.py [hunk 1]
```diff
@@ -68,13 +68,13 @@ def datetime(entry, option_key="Datetime", account=None, from_tz=None, **kwargs)
...
     try:
-        local = _dt.datetime.strptime(input, "%b %d %H:%M %Y")
+        local = _dt.datetime.strptime(entry, "%b %d %H:%M %Y")
     except ValueError:
         raise ValueError(
-            f"{option_key} must be entered in a 24-hour format such as: {now.strftime('%b %d %H:%H')}"
+            f"{option_key} must be entered in a 24-hour format such as: {now.strftime('%b %d %H:%M')}"
         )
     local_tz = from_tz.localize(local)
     return local_tz.astimezone(utc)
```
Comment (Henddher) [inline evennia/utils/validatorfuncs.py]: Line 48:
AccountDB is never used to retrieve the timezone.

File: evennia/utils/tests/test_validatorfuncs.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        # self.assertEqual(
+        #     datetime.timedelta(1+5*365, 2, 0, 0, 3, 4, 5),
+        #     validatorfuncs.duration('1d 2s 3m 4h 5w 5y'))
+
+    def test_duration_raises_ValueError(self):
+        for d in ['', '1', '5days', '1Week']:
+            with self.assertRaises(ValueError):
+                validatorfuncs.duration(d)
+
+    # def test_future_ok(self):
```
Comment (Henddher) [inline evennia/utils/tests/test_validatorfuncs.py]: There seems to be some issues with the creation of datetimes and their respective timezone. Some are coming up as naive.

I'll find the root cause and submit another PR with the fixes and these 2 tests enabled.

**Comment by Henddher** [comment]
Another one @Griatch :)

File: evennia/utils/tests/test_validatorfuncs.py (hunk 2 above)
Comment (Henddher) [reply evennia/utils/tests/test_validatorfuncs.py]: Root cause identified and addressed. All tests pass.

File: evennia/utils/tests/test_validatorfuncs.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+    def test_duration_ok(self):
+        for d in ['1d', '2w', '3h', '4s', '5m', '6y']:
+            self.assertTrue(
+                isinstance(validatorfuncs.duration(d), datetime.timedelta))
+
+        # THE FOLLOWING FAILS, year calculation seems to be incorrect
+        # self.assertEqual(
+        #     datetime.timedelta(1+5*365, 2, 0, 0, 3, 4, 5),
+        #     validatorfuncs.duration('1d 2s 3m 4h 5w 5y'))
```
[the remark this answers is not in the document]
Comment (Griatch) [reply evennia/utils/tests/test_validatorfuncs.py]: Interesting. This requires some further investigation into the duration validator.

File: evennia/utils/validatorfuncs.py (hunk 1 above)
Comment (Griatch) [reply evennia/utils/validatorfuncs.py]: It looks like the docstring was written first but the code did not properly implement the full functionality promised in the test. I don't recall off hand if you _can_ even get the timezone from the AccountDB object at this time ...

File: evennia/utils/tests/test_validatorfuncs.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Henddher) [reply evennia/utils/tests/test_validatorfuncs.py]: Filed bug #1970

File: evennia/utils/validatorfuncs.py (hunk 1 above)
Comment (Henddher) [reply evennia/utils/validatorfuncs.py]: Filed  bug #1971

**Comment by Henddher** [comment]
Great! :tada:

**Pull request merged**